lib/Compiler.js

```diff
@@ -186,35 +187,30 @@ class Compiler extends Tapable {
 			loader: null,
 			context: null
 		};
-		let deprecationReported = false;
 		this.parser = {
-			plugin: (hook, fn) => {
-				if(!deprecationReported) {
-					console.warn("webpack: Using compiler.parser is deprecated.\n" +
-						"Use compiler.plugin(\"compilation\", function(compilation, data) {\n  data.normalModuleFactory.plugin(\"parser\", function(parser, options) { parser.plugin(/* ... */); });\n}); instead. " +
-						"It was called " + new Error().stack.split("\n")[2].trim() + ".");
-					deprecationReported = true;
-				}
-				this.plugin("compilation", (compilation, data) => {
-					data.normalModuleFactory.plugin("parser", parser => {
-						parser.plugin(hook, fn);
+			plugin: util.deprecate(
+				(hook, fn) => {
+					this.plugin("compilation", (compilation, data) => {
+						data.normalModuleFactory.plugin("parser", parser => {
+							parser.plugin(hook, fn);
+						});
 					});
-				});
-			},
-			apply: () => {
-				const args = arguments;
-				if(!deprecationReported) {
-					console.warn("webpack: Using compiler.parser is deprecated.\n" +
-						"Use compiler.plugin(\"compilation\", function(compilation, data) {\n  data.normalModuleFactory.plugin(\"parser\", function(parser, options) { parser.apply(/* ... */); });\n}); instead. " +
-						"It was called " + new Error().stack.split("\n")[2].trim() + ".");
-					deprecationReported = true;
-				}
-				this.plugin("compilation", (compilation, data) => {
-					data.normalModuleFactory.plugin("parser", parser => {
-						parser.apply.apply(parser, args);
+				},
+				"webpack: Using compiler.parser is deprecated.\n" +
+				"Use compiler.plugin(\"compilation\", function(compilation, data) {\n  data.normalModuleFactory.plugin(\"parser\", function(parser, options) { parser.plugin(/* ... */); });\n}); instead. "
+			),
+			apply: util.deprecate(
+				() => {
+					const args = arguments;
+					this.plugin("compilation", (compilation, data) => {
+						data.normalModuleFactory.plugin("parser", parser => {
+							parser.apply(parser, args);
```

Comment:
No no a bug sneeked into this:

It should be `parser.apply.apply(parser, args);` instead of `parser.apply(parser, args);`

Could you send a new PR fixing this?